python/ray/air/checkpoint.py

```diff
@@ -246,6 +246,37 @@ def _metadata(self, metadata: _CheckpointMetadata):
         for attr, value in metadata.checkpoint_state.items():
             setattr(self, attr, value)
 
+    @property
+    def path(self) -> Optional[str]:
```

Comment:
Should `checkpoint.uri` get deprecated given this one?

Also, what if a user wants the local path for some reason when using cloud? Can only access through private property right now.

Reply:
I think we should keep URI to have counterparts to `to_uri` and `from_uri` for now. But I agree, we should probably see if we want to move to `to_path` and `from_path` instead.

Reply:
For accessing the local path, if a checkpoint has a URI, the checkpoint object does not have a local path stored. There may be cached data somewhere on the local node, but it doesn't have to be (e.g. if the checkpoint was stored on a worker node). The user can use `checkpoint.to_directory()` to download the data to a target directory.